mobile_cv/torch/utils_pytorch/shareables.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import logging
+import os
+import pickle
+from multiprocessing import shared_memory
+from typing import Any, List, Tuple, Union
+
+import mobile_cv.torch.utils_pytorch.comm as comm
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+class _SharedMemoryRef(object):
+    """Deal with the clean up of shared memory"""
+
+    def __init__(self, shm: shared_memory.SharedMemory, owner_pid: int):
+        self.shm = shm
+        self.owner_pid = owner_pid
+
+    def __del__(self):
+        self.shm.close()  # all instances should call close()
+        if os.getpid() == self.owner_pid:
+            self.shm.unlink()  # destroy the underlying shared memory block
```

Comment:
It looks like if master's refcount goes to 0 first, but workers are still using the data, bad memory access could happen.

Not sure if this could happen. Maybe it can happen if master throws an exception while other workers are still running.

Reply:
I read about this [comment](https://stackoverflow.com/questions/73544586/how-to-properly-close-and-unlink-shared-memory-of-multiprocessing) saying it is safe to call `shm.unlink()` before all handles are closed. I tried the following and it doesn't crash:
```
lst = SharedList([1, 2, 3])
time.sleep(comm.get_rank())  # mimic the workload that master process finishes first
print(lst[0])
del lst
```